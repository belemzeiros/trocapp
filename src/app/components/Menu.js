import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { withStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import MenuList from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Fade from '@material-ui/core/Fade';
import { Link } from 'react-router-dom';

const styles = theme => ({
  root: {
    width: '100%',
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit,
      width: 'auto',
    },
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
    width: '100%',
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 120,
      '&:focus': {
        width: 200,
      },
    },
  },
  menuLink: {
    '& a': {
      textDecoration: 'none',
      color: 'inherit',
    },
  },
});

class Menu extends Component {
  constructor(props) {
    super(props);

    this.state = {
      estaAberto: null,
    };
  }

  handleClick = event => {
    this.setState({ estaAberto: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ estaAberto: null });
  };

  render() {
    const { estaAberto } = this.state;
    const { classes, title } = this.props;
    const aberto = Boolean(estaAberto);

    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <IconButton
              className={classes.menuButton}
              color="inherit"
              aria-label="Menu"
              aria-owns={aberto ? 'fade-menu' : undefined}
              aria-haspopup="true"
              onClick={this.handleClick}
            >
              <MenuIcon />
            </IconButton>
            <MenuList
              id="fade-menu"
              anchorEl={estaAberto}
              open={aberto}
              onClose={this.handleClose}
              TransitionComponent={Fade}
            >
              <MenuItem className={classes.menuLink}>
                <Link to="/">Início</Link>
              </MenuItem>
              <MenuItem className={classes.menuLink}>
                <Link to="/sobre">Sobre Nós</Link>
              </MenuItem>
              <MenuItem className={classes.menuLink}>
                <Link to="/fale-conosco">Contato</Link>
              </MenuItem>
            </MenuList>
            <Typography
              className={classes.title}
              variant="title"
              color="inherit"
              noWrap
            >
              {title}
            </Typography>
            <div className={classes.grow} />
            <div className={classes.search} />
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

Menu.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  title: PropTypes.string.isRequired,
};

export default withStyles(styles)(Menu);
