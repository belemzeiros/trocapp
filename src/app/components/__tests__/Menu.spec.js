import React from 'react';
import { shallow, mount } from 'enzyme';
import Menu from '../Menu';

describe('menu', () => {
  it('deveria exibir o titulo da pagina', () => {
    const wrapper = shallow(<Menu title="Home" />);
    expect(wrapper.props().title).toEqual('Home');
  });
  it('deveria ser igual ao snapshot', () => {
    const wrapper = mount(<Menu title="home" />);
    expect(wrapper).toMatchSnapshot();
  });
});
