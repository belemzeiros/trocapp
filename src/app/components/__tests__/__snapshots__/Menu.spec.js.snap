// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu deveria ser igual ao snapshot 1`] = `
<WithStyles(Menu)
  title="home"
>
  <Menu
    classes={
      Object {
        "grow": "Menu-grow-2",
        "inputInput": "Menu-inputInput-8",
        "inputRoot": "Menu-inputRoot-7",
        "menuButton": "Menu-menuButton-3",
        "menuLink": "Menu-menuLink-9",
        "root": "Menu-root-1",
        "search": "Menu-search-5",
        "searchIcon": "Menu-searchIcon-6",
        "title": "Menu-title-4",
      }
    }
    title="home"
  >
    <div
      className="Menu-root-1"
    >
      <WithStyles(AppBar)
        position="static"
      >
        <AppBar
          classes={
            Object {
              "colorDefault": "MuiAppBar-colorDefault-15",
              "colorPrimary": "MuiAppBar-colorPrimary-16",
              "colorSecondary": "MuiAppBar-colorSecondary-17",
              "positionAbsolute": "MuiAppBar-positionAbsolute-12",
              "positionFixed": "MuiAppBar-positionFixed-11",
              "positionStatic": "MuiAppBar-positionStatic-14",
              "positionSticky": "MuiAppBar-positionSticky-13",
              "root": "MuiAppBar-root-10",
            }
          }
          color="primary"
          position="static"
        >
          <WithStyles(Paper)
            className="MuiAppBar-root-10 MuiAppBar-positionStatic-14 MuiAppBar-colorPrimary-16"
            component="header"
            elevation={4}
            square={true}
          >
            <Paper
              className="MuiAppBar-root-10 MuiAppBar-positionStatic-14 MuiAppBar-colorPrimary-16"
              classes={
                Object {
                  "elevation0": "MuiPaper-elevation0-20",
                  "elevation1": "MuiPaper-elevation1-21",
                  "elevation10": "MuiPaper-elevation10-30",
                  "elevation11": "MuiPaper-elevation11-31",
                  "elevation12": "MuiPaper-elevation12-32",
                  "elevation13": "MuiPaper-elevation13-33",
                  "elevation14": "MuiPaper-elevation14-34",
                  "elevation15": "MuiPaper-elevation15-35",
                  "elevation16": "MuiPaper-elevation16-36",
                  "elevation17": "MuiPaper-elevation17-37",
                  "elevation18": "MuiPaper-elevation18-38",
                  "elevation19": "MuiPaper-elevation19-39",
                  "elevation2": "MuiPaper-elevation2-22",
                  "elevation20": "MuiPaper-elevation20-40",
                  "elevation21": "MuiPaper-elevation21-41",
                  "elevation22": "MuiPaper-elevation22-42",
                  "elevation23": "MuiPaper-elevation23-43",
                  "elevation24": "MuiPaper-elevation24-44",
                  "elevation3": "MuiPaper-elevation3-23",
                  "elevation4": "MuiPaper-elevation4-24",
                  "elevation5": "MuiPaper-elevation5-25",
                  "elevation6": "MuiPaper-elevation6-26",
                  "elevation7": "MuiPaper-elevation7-27",
                  "elevation8": "MuiPaper-elevation8-28",
                  "elevation9": "MuiPaper-elevation9-29",
                  "root": "MuiPaper-root-18",
                  "rounded": "MuiPaper-rounded-19",
                }
              }
              component="header"
              elevation={4}
              square={true}
            >
              <header
                className="MuiPaper-root-18 MuiPaper-elevation4-24 MuiAppBar-root-10 MuiAppBar-positionStatic-14 MuiAppBar-colorPrimary-16"
              >
                <WithStyles(Toolbar)>
                  <Toolbar
                    classes={
                      Object {
                        "dense": "MuiToolbar-dense-48",
                        "gutters": "MuiToolbar-gutters-46",
                        "regular": "MuiToolbar-regular-47",
                        "root": "MuiToolbar-root-45",
                      }
                    }
                    disableGutters={false}
                    variant="regular"
                  >
                    <div
                      className="MuiToolbar-root-45 MuiToolbar-regular-47 MuiToolbar-gutters-46"
                    >
                      <WithStyles(IconButton)
                        aria-haspopup="true"
                        aria-label="Menu"
                        className="Menu-menuButton-3"
                        color="inherit"
                        onClick={[Function]}
                      >
                        <IconButton
                          aria-haspopup="true"
                          aria-label="Menu"
                          className="Menu-menuButton-3"
                          classes={
                            Object {
                              "colorInherit": "MuiIconButton-colorInherit-50",
                              "colorPrimary": "MuiIconButton-colorPrimary-51",
                              "colorSecondary": "MuiIconButton-colorSecondary-52",
                              "disabled": "MuiIconButton-disabled-53",
                              "label": "MuiIconButton-label-54",
                              "root": "MuiIconButton-root-49",
                            }
                          }
                          color="inherit"
                          disabled={false}
                          onClick={[Function]}
                        >
                          <WithStyles(ButtonBase)
                            aria-haspopup="true"
                            aria-label="Menu"
                            centerRipple={true}
                            className="MuiIconButton-root-49 MuiIconButton-colorInherit-50 Menu-menuButton-3"
                            disabled={false}
                            focusRipple={true}
                            onClick={[Function]}
                          >
                            <ButtonBase
                              aria-haspopup="true"
                              aria-label="Menu"
                              centerRipple={true}
                              className="MuiIconButton-root-49 MuiIconButton-colorInherit-50 Menu-menuButton-3"
                              classes={
                                Object {
                                  "disabled": "MuiButtonBase-disabled-56",
                                  "focusVisible": "MuiButtonBase-focusVisible-57",
                                  "root": "MuiButtonBase-root-55",
                                }
                              }
                              component="button"
                              disableRipple={false}
                              disableTouchRipple={false}
                              disabled={false}
                              focusRipple={true}
                              onClick={[Function]}
                              tabIndex="0"
                              type="button"
                            >
                              <button
                                aria-haspopup="true"
                                aria-label="Menu"
                                className="MuiButtonBase-root-55 MuiIconButton-root-49 MuiIconButton-colorInherit-50 Menu-menuButton-3"
                                disabled={false}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyUp={[Function]}
                                onMouseDown={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchMove={[Function]}
                                onTouchStart={[Function]}
                                tabIndex="0"
                                type="button"
                              >
                                <span
                                  className="MuiIconButton-label-54"
                                >
                                  <pure(Menu)>
                                    <Menu>
                                      <WithStyles(SvgIcon)>
                                        <SvgIcon
                                          classes={
                                            Object {
                                              "colorAction": "MuiSvgIcon-colorAction-61",
                                              "colorDisabled": "MuiSvgIcon-colorDisabled-63",
                                              "colorError": "MuiSvgIcon-colorError-62",
                                              "colorPrimary": "MuiSvgIcon-colorPrimary-59",
                                              "colorSecondary": "MuiSvgIcon-colorSecondary-60",
                                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit-64",
                                              "root": "MuiSvgIcon-root-58",
                                            }
                                          }
                                          color="inherit"
                                          component="svg"
                                          fontSize="default"
                                          viewBox="0 0 24 24"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            className="MuiSvgIcon-root-58"
                                            focusable="false"
                                            viewBox="0 0 24 24"
                                          >
                                            <g>
                                              <path
                                                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                                              />
                                            </g>
                                          </svg>
                                        </SvgIcon>
                                      </WithStyles(SvgIcon)>
                                    </Menu>
                                  </pure(Menu)>
                                </span>
                                <WithStyles(TouchRipple)
                                  center={true}
                                  innerRef={[Function]}
                                >
                                  <TouchRipple
                                    center={true}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child-69",
                                        "childLeaving": "MuiTouchRipple-childLeaving-70",
                                        "childPulsate": "MuiTouchRipple-childPulsate-71",
                                        "ripple": "MuiTouchRipple-ripple-66",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                                        "root": "MuiTouchRipple-root-65",
                                      }
                                    }
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      className="MuiTouchRipple-root-65"
                                      component="span"
                                      enter={true}
                                      exit={true}
                                    >
                                      <span
                                        className="MuiTouchRipple-root-65"
                                      />
                                    </TransitionGroup>
                                  </TouchRipple>
                                </WithStyles(TouchRipple)>
                              </button>
                            </ButtonBase>
                          </WithStyles(ButtonBase)>
                        </IconButton>
                      </WithStyles(IconButton)>
                      <WithStyles(Menu)
                        TransitionComponent={[Function]}
                        anchorEl={null}
                        id="fade-menu"
                        onClose={[Function]}
                        open={false}
                      >
                        <Menu
                          TransitionComponent={[Function]}
                          anchorEl={null}
                          classes={
                            Object {
                              "paper": "MuiMenu-paper-72",
                            }
                          }
                          disableAutoFocusItem={false}
                          id="fade-menu"
                          onClose={[Function]}
                          open={false}
                          theme={
                            Object {
                              "breakpoints": Object {
                                "between": [Function],
                                "down": [Function],
                                "keys": Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ],
                                "only": [Function],
                                "up": [Function],
                                "values": Object {
                                  "lg": 1280,
                                  "md": 960,
                                  "sm": 600,
                                  "xl": 1920,
                                  "xs": 0,
                                },
                                "width": [Function],
                              },
                              "direction": "ltr",
                              "mixins": Object {
                                "gutters": [Function],
                                "toolbar": Object {
                                  "@media (min-width:0px) and (orientation: landscape)": Object {
                                    "minHeight": 48,
                                  },
                                  "@media (min-width:600px)": Object {
                                    "minHeight": 64,
                                  },
                                  "minHeight": 56,
                                },
                              },
                              "overrides": Object {},
                              "palette": Object {
                                "action": Object {
                                  "active": "rgba(0, 0, 0, 0.54)",
                                  "disabled": "rgba(0, 0, 0, 0.26)",
                                  "disabledBackground": "rgba(0, 0, 0, 0.12)",
                                  "hover": "rgba(0, 0, 0, 0.08)",
                                  "hoverOpacity": 0.08,
                                  "selected": "rgba(0, 0, 0, 0.14)",
                                },
                                "augmentColor": [Function],
                                "background": Object {
                                  "default": "#fafafa",
                                  "paper": "#fff",
                                },
                                "common": Object {
                                  "black": "#000",
                                  "white": "#fff",
                                },
                                "contrastThreshold": 3,
                                "divider": "rgba(0, 0, 0, 0.12)",
                                "error": Object {
                                  "contrastText": "#fff",
                                  "dark": "#d32f2f",
                                  "light": "#e57373",
                                  "main": "#f44336",
                                },
                                "getContrastText": [Function],
                                "grey": Object {
                                  "100": "#f5f5f5",
                                  "200": "#eeeeee",
                                  "300": "#e0e0e0",
                                  "400": "#bdbdbd",
                                  "50": "#fafafa",
                                  "500": "#9e9e9e",
                                  "600": "#757575",
                                  "700": "#616161",
                                  "800": "#424242",
                                  "900": "#212121",
                                  "A100": "#d5d5d5",
                                  "A200": "#aaaaaa",
                                  "A400": "#303030",
                                  "A700": "#616161",
                                },
                                "primary": Object {
                                  "contrastText": "#fff",
                                  "dark": "#303f9f",
                                  "light": "#7986cb",
                                  "main": "#3f51b5",
                                },
                                "secondary": Object {
                                  "contrastText": "#fff",
                                  "dark": "#c51162",
                                  "light": "#ff4081",
                                  "main": "#f50057",
                                },
                                "text": Object {
                                  "disabled": "rgba(0, 0, 0, 0.38)",
                                  "hint": "rgba(0, 0, 0, 0.38)",
                                  "primary": "rgba(0, 0, 0, 0.87)",
                                  "secondary": "rgba(0, 0, 0, 0.54)",
                                },
                                "tonalOffset": 0.2,
                                "type": "light",
                              },
                              "props": Object {},
                              "shadows": Array [
                                "none",
                                "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
                                "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
                                "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
                                "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
                                "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
                                "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
                                "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
                                "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
                                "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
                                "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
                                "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
                                "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
                                "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
                                "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
                                "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
                                "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
                                "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
                                "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
                                "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
                                "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
                                "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
                                "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
                                "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
                                "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
                              ],
                              "shape": Object {
                                "borderRadius": 4,
                              },
                              "spacing": Object {
                                "unit": 8,
                              },
                              "transitions": Object {
                                "create": [Function],
                                "duration": Object {
                                  "complex": 375,
                                  "enteringScreen": 225,
                                  "leavingScreen": 195,
                                  "short": 250,
                                  "shorter": 200,
                                  "shortest": 150,
                                  "standard": 300,
                                },
                                "easing": Object {
                                  "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
                                  "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
                                  "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
                                  "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
                                },
                                "getAutoHeightDuration": [Function],
                              },
                              "typography": Object {
                                "body1": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "0.875rem",
                                  "fontWeight": 400,
                                  "lineHeight": "1.46429em",
                                },
                                "body2": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "0.875rem",
                                  "fontWeight": 500,
                                  "lineHeight": "1.71429em",
                                },
                                "button": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "0.875rem",
                                  "fontWeight": 500,
                                  "textTransform": "uppercase",
                                },
                                "caption": Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "0.75rem",
                                  "fontWeight": 400,
                                  "lineHeight": "1.375em",
                                },
                                "display1": Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "2.125rem",
                                  "fontWeight": 400,
                                  "lineHeight": "1.20588em",
                                },
                                "display2": Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "2.8125rem",
                                  "fontWeight": 400,
                                  "lineHeight": "1.13333em",
                                  "marginLeft": "-.02em",
                                },
                                "display3": Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "3.5rem",
                                  "fontWeight": 400,
                                  "letterSpacing": "-.02em",
                                  "lineHeight": "1.30357em",
                                  "marginLeft": "-.02em",
                                },
                                "display4": Object {
                                  "color": "rgba(0, 0, 0, 0.54)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "7rem",
                                  "fontWeight": 300,
                                  "letterSpacing": "-.04em",
                                  "lineHeight": "1.14286em",
                                  "marginLeft": "-.04em",
                                },
                                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                "fontSize": 14,
                                "fontWeightLight": 300,
                                "fontWeightMedium": 500,
                                "fontWeightRegular": 400,
                                "headline": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "1.5rem",
                                  "fontWeight": 400,
                                  "lineHeight": "1.35417em",
                                },
                                "pxToRem": [Function],
                                "round": [Function],
                                "subheading": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "1rem",
                                  "fontWeight": 400,
                                  "lineHeight": "1.5em",
                                },
                                "title": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                                  "fontSize": "1.3125rem",
                                  "fontWeight": 500,
                                  "lineHeight": "1.16667em",
                                },
                              },
                              "zIndex": Object {
                                "appBar": 1100,
                                "drawer": 1200,
                                "mobileStepper": 1000,
                                "modal": 1300,
                                "snackbar": 1400,
                                "tooltip": 1500,
                              },
                            }
                          }
                          transitionDuration="auto"
                        >
                          <WithStyles(Popover)
                            PaperProps={
                              Object {
                                "classes": Object {
                                  "root": "MuiMenu-paper-72",
                                },
                              }
                            }
                            TransitionComponent={[Function]}
                            anchorEl={null}
                            anchorOrigin={
                              Object {
                                "horizontal": "left",
                                "vertical": "top",
                              }
                            }
                            getContentAnchorEl={[Function]}
                            id="fade-menu"
                            onClose={[Function]}
                            onEnter={[Function]}
                            open={false}
                            transformOrigin={
                              Object {
                                "horizontal": "left",
                                "vertical": "top",
                              }
                            }
                            transitionDuration="auto"
                          >
                            <Popover
                              PaperProps={
                                Object {
                                  "classes": Object {
                                    "root": "MuiMenu-paper-72",
                                  },
                                }
                              }
                              TransitionComponent={[Function]}
                              anchorEl={null}
                              anchorOrigin={
                                Object {
                                  "horizontal": "left",
                                  "vertical": "top",
                                }
                              }
                              anchorReference="anchorEl"
                              classes={
                                Object {
                                  "paper": "MuiPopover-paper-73",
                                }
                              }
                              elevation={8}
                              getContentAnchorEl={[Function]}
                              id="fade-menu"
                              marginThreshold={16}
                              onClose={[Function]}
                              onEnter={[Function]}
                              open={false}
                              transformOrigin={
                                Object {
                                  "horizontal": "left",
                                  "vertical": "top",
                                }
                              }
                              transitionDuration="auto"
                            >
                              <WithStyles(Modal)
                                BackdropProps={
                                  Object {
                                    "invisible": true,
                                  }
                                }
                                id="fade-menu"
                                onClose={[Function]}
                                open={false}
                              >
                                <Modal
                                  BackdropComponent={[Function]}
                                  BackdropProps={
                                    Object {
                                      "invisible": true,
                                    }
                                  }
                                  classes={
                                    Object {
                                      "hidden": "MuiModal-hidden-75",
                                      "root": "MuiModal-root-74",
                                    }
                                  }
                                  disableAutoFocus={false}
                                  disableBackdropClick={false}
                                  disableEnforceFocus={false}
                                  disableEscapeKeyDown={false}
                                  disablePortal={false}
                                  disableRestoreFocus={false}
                                  hideBackdrop={false}
                                  id="fade-menu"
                                  keepMounted={false}
                                  manager={
                                    ModalManager {
                                      "containers": Array [],
                                      "data": Array [],
                                      "handleContainerOverflow": true,
                                      "hideSiblingNodes": true,
                                      "modals": Array [],
                                    }
                                  }
                                  onClose={[Function]}
                                  open={false}
                                />
                              </WithStyles(Modal)>
                            </Popover>
                          </WithStyles(Popover)>
                        </Menu>
                      </WithStyles(Menu)>
                      <WithStyles(Typography)
                        className="Menu-title-4"
                        color="inherit"
                        noWrap={true}
                        variant="title"
                      >
                        <Typography
                          align="inherit"
                          className="Menu-title-4"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter-89",
                              "alignJustify": "MuiTypography-alignJustify-91",
                              "alignLeft": "MuiTypography-alignLeft-88",
                              "alignRight": "MuiTypography-alignRight-90",
                              "body1": "MuiTypography-body1-85",
                              "body2": "MuiTypography-body2-84",
                              "button": "MuiTypography-button-87",
                              "caption": "MuiTypography-caption-86",
                              "colorError": "MuiTypography-colorError-100",
                              "colorInherit": "MuiTypography-colorInherit-95",
                              "colorPrimary": "MuiTypography-colorPrimary-96",
                              "colorSecondary": "MuiTypography-colorSecondary-97",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary-98",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary-99",
                              "display1": "MuiTypography-display1-80",
                              "display2": "MuiTypography-display2-79",
                              "display3": "MuiTypography-display3-78",
                              "display4": "MuiTypography-display4-77",
                              "gutterBottom": "MuiTypography-gutterBottom-93",
                              "headline": "MuiTypography-headline-81",
                              "noWrap": "MuiTypography-noWrap-92",
                              "paragraph": "MuiTypography-paragraph-94",
                              "root": "MuiTypography-root-76",
                              "subheading": "MuiTypography-subheading-83",
                              "title": "MuiTypography-title-82",
                            }
                          }
                          color="inherit"
                          gutterBottom={false}
                          headlineMapping={
                            Object {
                              "body1": "p",
                              "body2": "aside",
                              "display1": "h1",
                              "display2": "h1",
                              "display3": "h1",
                              "display4": "h1",
                              "headline": "h1",
                              "subheading": "h3",
                              "title": "h2",
                            }
                          }
                          noWrap={true}
                          paragraph={false}
                          variant="title"
                        >
                          <h2
                            className="MuiTypography-root-76 MuiTypography-title-82 MuiTypography-colorInherit-95 MuiTypography-noWrap-92 Menu-title-4"
                          >
                            home
                          </h2>
                        </Typography>
                      </WithStyles(Typography)>
                      <div
                        className="Menu-grow-2"
                      />
                      <div
                        className="Menu-search-5"
                      />
                    </div>
                  </Toolbar>
                </WithStyles(Toolbar)>
              </header>
            </Paper>
          </WithStyles(Paper)>
        </AppBar>
      </WithStyles(AppBar)>
    </div>
  </Menu>
</WithStyles(Menu)>
`;
